Bottom: 9e1add009f1c4d157ef031871290abe1c805733a
Top:    fb14f724582dcf21efefd486a3cbe180478be85d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-04 17:09:51 +0000

Refresh of board-outline-polygon-generati

---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index fac4e56..be37956 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -553,13 +553,17 @@ int compute_depth (int group)
       depth = (10 - 3) * 200 / gport->zoom;
     }
   } else if (SL_TYPE (idx) == SL_SILK) {
-    if (SL_SIDE (idx) == SL_TOP_SIDE && !Settings.ShowSolderSide) {
+    if (SL_SIDE (idx) == SL_TOP_SIDE) {
       depth = (max_depth + 5) * 200 / gport->zoom;
     } else {
       depth = (10 - 5) * 200 / gport->zoom;
     }
   } else if (SL_TYPE (idx) == SL_INVISIBLE) {
-    depth = (10 - 3) * 200 / gport->zoom;
+    if (Settings.ShowSolderSide) {
+      depth = (max_depth + 5) * 200 / gport->zoom;
+    } else {
+      depth = (10 - 5) * 200 / gport->zoom;
+    }
   }
 
   last_depth_computed = depth;
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 7552493..82098ca 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1542,6 +1542,9 @@ ghid_draw_everything (BoxTypePtr drawn_area)
   if (!global_view_2d && save_show_solder)
     reverse_layers = !reverse_layers;
 
+  PCB->Data->SILKLAYER.Color = PCB->ElementColor;
+  PCB->Data->BACKSILKLAYER.Color = PCB->InvisibleObjectsColor;
+
   solder_group = GetLayerGroupNumberByNumber (max_layer + SOLDER_LAYER);
   component_group = GetLayerGroupNumberByNumber (max_layer + COMPONENT_LAYER);
 
@@ -1597,6 +1600,7 @@ ghid_draw_everything (BoxTypePtr drawn_area)
         SWAP_IDENT = save_swap;
         gui->set_layer (NULL, SL (FINISHED, 0), 0);
       }
+      gui->set_layer ("invisible", SL (INVISIBLE, 0), 0);
     }
 #endif
     if (global_view_2d)
